illwill.nim

```diff
@@ -658,49 +658,53 @@ else:  # OS X & Linux
     else:
       gMouseInfo.scrollDir = ScrollDirection.sdNone
 
-  proc parseKey(charsRead: int): Key =
+  proc parseKeys(charsRead: int): seq[Key] =
     # Inspired by
     # https://github.com/mcandre/charm/blob/master/lib/charm.c
-    var key = Key.None
     if charsRead == 1:
       let ch = keyBuf[0]
       case ch:
-      of   9: key = Key.Tab
-      of  10: key = Key.Enter
-      of  27: key = Key.Escape
-      of  32: key = Key.Space
-      of 127: key = Key.Backspace
+      of   9: result.add Key.Tab
+      of  10: result.add Key.Enter
+      of  27: result.add Key.Escape
+      of  32: result.add Key.Space
+      of 127: result.add Key.Backspace
       of 0, 29, 30, 31: discard   # these have no Windows equivalents so
                                   # we'll ignore them
       else:
-        key = toKey(ch)
+        result.add toKey(ch)
 
     elif charsRead > 3 and keyBuf[0] == 27 and keyBuf[1] == 91 and keyBuf[2] == 60: # TODO what are these :)
       fillGlobalMouseInfo(keyBuf)
-      return Key.Mouse
+      result.add Key.Mouse
 
     else:
       var inputSeq = ""
       for i in 0..<charsRead:
         inputSeq &= char(keyBuf[i])
-      for keyCode, sequences in keySequences.pairs:
-        for s in sequences:
-          if s == inputSeq:
-            key = toKey(keyCode)
-    result = key
-
-  proc getKeyAsync(): Key =
+      var off = 0
+      while off < inputSeq.len:
+        block found:
+          for keyCode, sequences in keySequences.pairs:
+            for s in sequences:
+              if off + s.len <= inputSeq.len and s == inputSeq[off..off+s.high]:
+                result.add toKey(keyCode)
+                off += s.len
+                break found
+          if off < inputSeq.len:
+            result.add toKey(int(inputSeq[off]))
```

Comment:
minor: Can you follow my code style and use parenthesie around the arguments everywhere?